packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -0,0 +1,267 @@
+}
+
+// tslint:disable-next-line:no-any
+function extractArchitectConfig(_config: any): JsonObject {
+  return {};
+}
+// tslint:disable-next-line:no-any
+function extractProjectsConfig(config: any): any {
+  const builderPackage = '@angular-devkit/build-webpack';
+  // convert the apps to projects
+  // tslint:disable-next-line:no-any
+  const projectMap = config.apps.map((app: any, idx: number) => {
+    const name = app.name || `$$proj${idx}`;
+    // tslint:disable-next-line:no-any
+    const project: any = {
+      root: app.root,
+      projectType: 'application',
+      architect: {},
+    };
+    // tslint:disable-next-line:no-any
+    const architect = project.architect;
+
+    const extraEntryMapper = (extraEntry: string | JsonObject) =>
+      typeof extraEntry === 'string' ? { input: extraEntry } : extraEntry;
+
+    // Browser target
+    const buildOptions: JsonObject = {
+      // Make outputPath relative to root.
+      outputPath: relative(normalize('/' + app.root), normalize('/' + app.outDir + '/' + name)),
+      index: app.index,
+      main: app.main,
+      polyfills: app.polyfills,
+      tsConfig: app.tsconfig,
+      progress: false,
+    };
+
+    buildOptions.scripts = (app.scripts || []).map(extraEntryMapper);
+    buildOptions.styles = (app.styles || []).map(extraEntryMapper);
+
+    buildOptions.assets = (app.assets || []).map((asset: string | JsonObject) =>
+      typeof asset === 'string' ? { glob: asset } : asset);
+
+    architect['build'] = {
+      builder: `${builderPackage}:build`,
+      options: buildOptions,
+      configurations: {
+        production: {
+          optimizationLevel: 1,
+          outputHashing: 'all',
+          sourceMap: false,
+          extractCss: true,
+          namedChunks: false,
+          aot: true,
+          extractLicenses: true,
+          vendorChunk: false,
+          buildOptimizer: true,
+        },
+      },
+    };
+
+    // Serve target
+    const serveOptions: JsonObject = {
+      browserTarget: `${name}:build`,
+    };
+
+    architect['serve'] = {
+      builder: `${builderPackage}:dev-server`,
+      options: serveOptions,
+      configurations: {
+        production: {
+          browserTarget: `${name}:build`,
+        },
+      },
+    };
+
+    // Test target
+    const testOptions: JsonObject = {
+      main: app.test,
+      polyfills: app.polyfills,
+      tsConfig: app.testTsconfig,
+      // Make karmaConfig relative to root.
+      karmaConfig: relative(normalize('/' + app.root), normalize('/' + config.test.karma.config)),
+    };
+
+    testOptions.scripts = (app.scripts || []).map(extraEntryMapper);
+    testOptions.styles = (app.styles || []).map(extraEntryMapper);
+
+    testOptions.assets = (app.assets || []).map((asset: string | JsonObject) =>
+      typeof asset === 'string' ? { glob: asset } : asset);
+
+    architect['test'] = {
+      builder: `${builderPackage}:karma`,
+      options: testOptions,
+    };
+
+    // // Protractor target
```

Comment:
The protractor target still needs to added in a separate app:

https://github.com/filipesilva/workspace-project/blob/17b36374b76112908348c6bb08b434420e6cffc3/.workspace.json#L112-L133